bdai_ros2_wrappers/bdai_ros2_wrappers/action_client.py

```diff
@@ -1,58 +1,33 @@
 # Copyright (c) 2023 Boston Dynamics AI Institute Inc.  All rights reserved.
-from typing import Callable, Optional, Type, Union
+from typing import Callable, Optional, Type
 
 import rclpy.action
-from rclpy import Context
+from rclpy.node import Node
 
+import bdai_ros2_wrappers.process as process
 from bdai_ros2_wrappers.action_handle import ActionHandle
-from bdai_ros2_wrappers.node import NodeWrapper
 from bdai_ros2_wrappers.type_hints import Action
 
 
 class ActionClientWrapper(rclpy.action.ActionClient):
     """A wrapper for ros2's ActionClient for extra functionality"""
 
-    def __init__(
-        self,
-        action_type: Type[Action],
-        action_name: str,
-        node: Union[str, NodeWrapper],
-        *,
-        namespace: Optional[str] = None,
-        context: Optional[Context] = None,
-        num_executor_threads: int = 1,
-    ) -> None:
+    def __init__(self, action_type: Type[Action], action_name: str, node: Optional[Node] = None) -> None:
         """Constructor
 
         Args:
             action_type (Type[Action]): The type of the action
             action_name (str): The name of the action (for logging purposes)
-            node (Union[str, NodeWrapper]): The name of the internal node if a new one is to be created or a NodeWrapper
-                if this object should share the node
-            namespace (Optional[str]): The namespace for the internal node
-            context (Optional[Context]): The context for the internal node
-            num_executor_threads (int): Number of threads the executor should use.
-                1 => Single Thread, >1 => Multithread of that number of executors,
-                -1 => Multithread with as many threads as the system can do using `multiprocessing.cpu_count`
+            node (Optional[Node]): optional node for action client, defaults to the current process node
         """
-        if isinstance(node, NodeWrapper):
-            self._node_wrapper = node
-        elif isinstance(node, str):
-            self._node_wrapper = NodeWrapper(
-                f"{node}_{action_name}_client_wrapper_node",
-                namespace=namespace,
-                context=context,
-                spin_thread=True,
-                num_executor_threads=num_executor_threads,
-            )
-        else:
-            raise TypeError("`node` should be either a `NodeWrapper` or `str`")
-        super().__init__(self._node_wrapper, action_type, action_name)
-        self._node_wrapper.get_logger().info(
-            f"Waiting for action server for {self._node.get_namespace()}/{action_name}"
-        )
+        node = node or process.node()
+        if node is None:
+            raise ValueError("no ROS 2 node available")
```

Comment:
Should this perhaps tell the user a little more - like "To use the default process node make sure you are using X executor" or something?

Reply:
Note however that this idea of a default process node is orthogonal to that of the executor. We could be using a single-threaded executor in the background (with all the limitations you already know about).

Reply:
That's fair. I agree with Jenny that we should have a suggestion for how to fix the error. Something like "please provide a node or use the process-wide node via _".

Reply:
See https://github.com/bdaiinstitute/ros_utilities/pull/25/commits/30ad62da27fefa15637b9dc2063a5aa4b5c2f3f6.